packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -0,0 +1,352 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/physics.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'scroll_metrics.dart';
+import 'scroll_notification.dart';
+import 'scroll_position.dart';
+import 'ticker_provider.dart';
+
+abstract class ScrollActivityDelegate {
+  AxisDirection get axisDirection;
+
+  double get pixels;
+  double setPixels(double pixels);
+  double applyUserOffset(double delta);
+
+  void goIdle();
+  void goBallistic(double velocity);
+}
+
+/// Base class for scrolling activities like dragging and flinging.
+///
+/// See also:
+///
+///  * [ScrollIndependentPosition], which uses [ScrollActivity] objects to
```

Comment:
there's a bunch of mentions of ScrollIndependentPosition and ScrollPosition in this file